libcaf_core/caf/message_builder.hpp

```diff
@@ -91,7 +93,7 @@ class CAF_CORE_EXPORT message_builder {
   /// Converts the buffer to an actual message object and transfers
   /// ownership of the data to it, leaving this object in an invalid state.
   /// @warning Calling *any*  member function on this object afterwards
-  ///          is undefined behavior (dereferencing a `nullptr`)
+  ///          is undefined behavior.
```

Comment:
Wouldn't it make sense to allow calling `clear()`?

Reply:
I don't think we should support any operation on a moved-from object in general. What use case would require clearing moved-from objects instead of simply creating new builders?

Supporting `clear` would require the `type_id_list_builder` to reset its members in `move_to_list`. That would be dead code 99% of the time and only add a maintenance headache: one (weird) self-incurred requirement we need to track, test, and continue to support.